packages/treeview/src/useTreeview.ts

```diff
@@ -84,167 +65,145 @@ export function useTreeview<Item = any>({
 
     onChange(newValue);
+
     if (!isControlled) {
       setOpenNodesState(newValue);
     }
   };
 
-  const getTreeProps = ({ label, role = 'tree', onFocus, ...props }: IGetTreeProps = {}) => {
-    if (role !== 'tree') {
-      throw new Error(
-        `Accessibility Error: "role" prop must have value "tree" in "getTreeProps()"`
-      );
-    }
-
-    return {
-      role,
-      'aria-label': label,
-      'aria-orientation': orientation,
-      'data-garden-container-id': 'containers.treeview',
-      'data-garden-container-version': PACKAGE_VERSION,
-      onFocus,
-      ...props
-    };
-  };
-
-  const getNodeProps = (
-    {
-      item,
-      role = 'treeitem',
-      nodeType = 'end',
-      onClick,
-      onFocus,
-      onKeyDown,
-      focusRef,
-      ...props
-    }: IGetNodeProps<Item> = {} as any
-  ) => {
-    requiredArguments(item, 'item', 'getNodeProps');
-    requiredArguments(focusRef, 'focusRef', 'getNodeProps');
-
-    if (role !== 'treeitem') {
-      throw new Error(
-        `Accessibility Error: "role" prop must have value "treeitem" in "getNodeProps()"`
-      );
-    }
-
-    if (nodeType !== 'end' && nodeType !== 'parent') {
-      throw new Error(
-        `Accessibility Error: "nodeType" prop value must be either "parent" or "end" in "getNodeProps()"`
-      );
-    }
+  const getTreeProps: IUseTreeviewReturnValue<Item>['getTreeProps'] = ({
+    role = 'tree',
+    ...other
+  }) => ({
+    ...getContainerProps(other),
+    role: role === null ? undefined : role,
+    'aria-orientation': orientation,
+    'data-garden-container-id': 'containers.treeview',
+    'data-garden-container-version': PACKAGE_VERSION
+  });
 
+  const getNodeProps: IUseTreeviewReturnValue<Item>['getNodeProps'] = ({
+    item,
+    role = 'treeitem',
+    nodeType = 'end',
+    onClick,
+    onFocus,
+    onKeyDown,
+    focusRef,
+    ...other
+  }) => {
     const expanded = nodeType === 'parent' ? isNodeExpanded(item) : undefined;
 
-    const onClickFn = composeEventHandlers(onClick, (e: MouseEvent) => {
+    const handleClick = composeEventHandlers(onClick, (event: MouseEvent) => {
       if (nodeType === 'parent') {
-        e.preventDefault();
+        event.preventDefault();
         toggleParent(item!);
       } else {
-        e.stopPropagation();
+        event.stopPropagation();
       }
     });
 
-    return {
-      role,
-      item,
-      focusRef,
-      'aria-expanded': expanded,
-      tabIndex: ownFocusedItem === undefined || ownFocusedItem === item ? 0 : -1,
-      'data-garden-container-version': PACKAGE_VERSION,
-      onClick: onClickFn,
-      onFocus: composeEventHandlers(onFocus, (event: Event) => {
-        event.stopPropagation();
-        setOwnFocusedItem(item);
-      }),
-      onKeyDown: composeEventHandlers(onKeyDown, (event: KeyboardEvent): void => {
-        const target = focusRef.current as HTMLElement;
+    const handleFocus: FocusEventHandler = event => {
```

Comment:
Should we extend this with `composeEventHandlers`?

Reply:
It is below the the `onFocus` return value.

Reply:
Gotcha!